Templates/Multiplayer/Template/Gem/Code/Source/AutoGen/NetworkRandomComponent.AutoComponent.xml

```diff
@@ -9,6 +9,6 @@
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:noNamespaceSchemaLocation="MultiplayerAutoComponentSchema.xsd">
 
-    <NetworkProperty Type="uint64_t" Init="0" Name="Seed" ReplicateFrom="Authority" ReplicateTo="Client" Container="Object" IsPublic="true" IsRewindable="false" IsPredictable="false" ExposeToEditor="false" ExposeToScript="false" GenerateEventBindings="false" Description="The RNG seed" />
+    <NetworkProperty Type="uint64_t" Init="0" Name="Seed" ReplicateFrom="Authority" ReplicateTo="Client" Container="Object" IsPublic="true" IsRewindable="false" IsPredictable="true" ExposeToEditor="false" ExposeToScript="false" GenerateEventBindings="false" Description="The RNG seed" />
```

Comment:
It’s predictable to allow players to generate random numbers. This is useful if the player shoots a weapon and random recoil needs to be applied. The player can successfully modify the random number generator locally and use it instantly, and it can assume the server will use that same random number when it processes the weapon shot.

Reply:
That's exactly right. The RNG is based on a seed that is synchronized. It should be the case that the same numbers are produced if the same requests for a random number are produced in the same order on both endpoints. This is a fairly typical approach to RNG between a client and server with corrections acting as a mechanism to resync if the sequence isn't properly mirrored.

As far as why it needed to be predictable, there were Client rolls and the split makes it so that non-predictable properties can't be set on a role that is read only.

Reply:
ah ha! thanks to you both for the clarification.